tests/Component.Tests/EventStreamTests.cs

```diff
@@ -0,0 +1,57 @@
+﻿using Component.Tests.Fakes;
+using Components;
+using FluentAssertions;
+using Microsoft.Framework.DependencyInjection;
+using System;
+using Xunit;
+
+namespace Component.Tests
+{
+    public class EventStreamTests
+    {
+        static IServiceProvider CreateProvider() => new ServiceCollection().AddInMemoryEventStream().BuildServiceProvider();
+        static EventStream CreateEventStream(IServiceProvider serviceProvider = null) => (serviceProvider ?? CreateProvider()).GetRequiredService<EventStream>();
+
+        [Fact]
+        public static void GivenANullObserverBasedSubscriberThrowArgumentNullException()
+        {
+            var stream = CreateEventStream();
+
+            Action act = () => stream.Subscribe(default(IObserver<TestEvent>));
+
+            act.ShouldThrow<ArgumentNullException>();
+        }
+
+        [Fact]
+        public static void GivenANullActionBasedSubscriberThrowArgumentNullException()
+        {
+            var stream = CreateEventStream();
```

Comment:
Spacing on these methods. Although I don't write the A/A/A comments like some others, I use the line spaces to distinguish the 3 sections. i.e. Keep the Arrange statements together and separate with a line from the Act, likewise for the Assert. In this case, the Act and Assert are on the same (last) line. In either case, the two Arrange statements can be collapsed into a single line.